core/Request.php

```diff
@@ -123,7 +141,13 @@ public function __construct()
         $this->cookie = $this->_normalize($_COOKIE);
         $this->files = $this->_normalize($_FILES);
         $this->server = $this->_normalize($_SERVER);
-
+        if($this->server('REQUEST_METHOD') === 'PUT') {
+            parse_str(file_get_contents('php://input', "r"), $this->put);
```

Comment:
This won't work.  parse_str() has no application here unless a query string were sent in the raw body of the request where it will more often be json (which you can't get using $this->request->put() from a controller this way) and form data psychobabble.  This guarantees you can't receive PUT JSON body content in your controller (or anywhere for that matter) if that's what's sent.